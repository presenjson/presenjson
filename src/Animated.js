import React from 'react';

const def = {
    approach: '7s ease-out',
    land: '5s cubic-bezier(0,1,.15,.84)',
    fadeInOut: '3s ease-in-out',
    fadeOut: '3s ease-in-out',
    blur: '3s ease-in-out'
};

export default ({
    children,
    className = '',
    fillMode = 'forwards',
    direction = 'normal',
    count = 'initial',
    delay = '0s',
    ...rest
}) => {
    const style = {
        animation: Object.entries(rest)
            .map(
                (o) => `${o[0]} ${(o[1] !== true && o[1]) || def[o[0]] || '0s'}`
            )
            .join(', '),
        animationFillMode: fillMode,
        animationDirection: direction,
        animationIterationCount: count,
        animationDelay: delay
    };
    return (
        <div className={`fx-container ${className}`} style={style}>
            {children}
        </div>
    );
};
