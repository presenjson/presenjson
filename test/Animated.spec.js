import React from 'react';
import { shallow } from 'enzyme';
import Animated from '../src/Animated';

describe('Animated', () => {
    it('renders correctly', () => {
        expect(
            shallow(
                <Animated>
                    <div className="test" />
                </Animated>
            )
        ).toMatchSnapshot();
    });
    it('sets animation name, fill-mode, delay, count', () => {
        const animated = shallow(
            <Animated
                testAnimation="30s"
                otherAnimation="20s"
                fillMode="foo"
                count="bar"
                delay="3s"
                direction="baz"
                className="the-class"
            />
        );
        expect(animated).toMatchSnapshot();
    });
    it('uses default animation lengths and easing functions for animations', () => {
        const animated = shallow(
            <Animated approach fadeInOut fadeOut blur land />
        );
        expect(animated).toMatchSnapshot();
    });
});
