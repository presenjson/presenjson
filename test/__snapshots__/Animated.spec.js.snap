// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animated renders correctly 1`] = `
<div
  className="fx-container "
  style={
    Object {
      "animation": "",
      "animationDelay": "0s",
      "animationDirection": "normal",
      "animationFillMode": "forwards",
      "animationIterationCount": "initial",
    }
  }
>
  <div
    className="test"
  />
</div>
`;

exports[`Animated sets animation name, fill-mode, delay, count 1`] = `
<div
  className="fx-container the-class"
  style={
    Object {
      "animation": "testAnimation 30s, otherAnimation 20s",
      "animationDelay": "3s",
      "animationDirection": "baz",
      "animationFillMode": "foo",
      "animationIterationCount": "bar",
    }
  }
/>
`;

exports[`Animated uses default animation lengths and easing functions for animations 1`] = `
<div
  className="fx-container "
  style={
    Object {
      "animation": "approach 7s ease-out, fadeInOut 3s ease-in-out, fadeOut 3s ease-in-out, blur 3s ease-in-out, land 5s cubic-bezier(0,1,.15,.84)",
      "animationDelay": "0s",
      "animationDirection": "normal",
      "animationFillMode": "forwards",
      "animationIterationCount": "initial",
    }
  }
/>
`;
